### PR TITLE
[fix](update) Update set value should consider sequence column

### PR DESCRIPTION
**Pull request by hust-hhb**
## Proposed changes
When using update command to set column value, if the column is sequence column, the column '__DORIS_SEQUENCE_COL__' should also be set to the same value.
Here is an example:
            CREATE TABLE IF NOT EXISTS table1(
                user_id bigint,
                date1 date,
                group_id bigint
            )
            UNIQUE KEY(user_id)
            DISTRIBUTED BY HASH(user_id) BUCKETS 5 properties(
                "function_column.sequence_col"='group_id',
                "replication_num" = "1",
                "in_memory"="false"
            );
insert into table1 values(1,'20240131',100);
insert into table1 values(2,'20240131',100);
now select * from table , the result is
1	2024-01-31	100
2	2024-01-31	100

UPDATE table1  SET group_id=200;
now select * from table , the result is
1	2024-01-31	200
2	2024-01-31	200

insert into table1 values(2,'20240131',100);
now select * from table , the result is, which is wrong
1	2024-01-31	200
2	2024-01-31	100

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hust-hhb** [comment]
run buildall

**Comment by hust-hhb** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/analysis/UpdateStmt.java [hunk 1]
```diff
@@ -211,12 +212,21 @@ private void analyzeSetExprs(Analyzer analyzer) throws AnalysisException {
...
+                    .filter(col -> col.getName().equalsIgnoreCase(olapTable.getSequenceMapCol())).findFirst();
+        }
         for (Column column : targetTable.getColumns()) {
             Expr expr = new SlotRef(targetTableRef.getAliasAsName(), column.getName());
             boolean existInExpr = false;
             for (BinaryPredicate setExpr : setExprs) {
                 Expr lhs = setExpr.getChild(0);
-                if (((SlotRef) lhs).getColumn().equals(column)) {
+                Column exprColumn = ((SlotRef) lhs).getColumn();
+                if (exprColumn.equals(column) || (olapTable.hasSequenceCol()
```
Comment (zhannngchen) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/UpdateStmt.java:227]: Add comment here

**Review by zhannngchen — CHANGES_REQUESTED** [review]
consider Nereids as well

**Comment by hust-hhb** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/analysis/UpdateStmt.java (hunk 1 above)
Comment (hust-hhb) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/UpdateStmt.java:230]: @zhannngchen already change the code, take a look plz.

**Comment by hust-hhb** [comment]
run buildall

File: regression-test/suites/update/test_update.groovy [hunk 2]
```diff
@@ -15,6 +15,39 @@
 // specific language governing permissions and limitations
 // under the License.
 suite("test_update", "p0") {
-    // todo:test update
-    sql "show tables"
+    def tbName = "test_update"
+    sql "DROP TABLE IF EXISTS ${tbName}"
```
[the remark this answers is not in the document]
Comment (hust-hhb) [reply regression-test/suites/update/test_update.groovy]: @zhannngchen done

**Comment by hust-hhb** [comment]
run buildall

**Review by zhannngchen — DISMISSED** [review]
LGTM

**Comment by hust-hhb** [comment]
run buildall

**Review by zhannngchen — APPROVED** [review]

**Review by morrySnow — APPROVED** [review]

**Review by liaoxin01 — APPROVED** [review]

**Pull request merged**